worker/src/RTC/RtpStreamSend.cpp

```diff
@@ -14,30 +14,141 @@ namespace RTC
 	static constexpr size_t MaxRequestedPackets{ 17 };
 	thread_local static std::vector<RTC::RtpStreamSend::StorageItem*> RetransmissionContainer(
 	  MaxRequestedPackets + 1);
-	// Don't retransmit packets older than this (ms).
-	static constexpr uint32_t MaxRetransmissionDelay{ 2000 };
 	static constexpr uint32_t DefaultRtt{ 100 };
+	static constexpr uint16_t MaxSeq = std::numeric_limits<uint16_t>::max();
 
-	static void resetStorageItem(RTC::RtpStreamSend::StorageItem* storageItem)
+	void RtpStreamSend::StorageItem::Reset()
 	{
 		MS_TRACE();
 
-		MS_ASSERT(storageItem, "storageItem cannot be nullptr");
+		this->packet.reset();
+		this->ssrc           = 0;
+		this->sequenceNumber = 0;
+		this->timestamp      = 0;
+		this->resentAtMs     = 0;
+		this->sentTimes      = 0;
+	}
+
+	RtpStreamSend::StorageItem* RtpStreamSend::StorageItemBuffer::GetFirst() const
+	{
+		return this->Get(this->startSeq);
+	}
+
+	RtpStreamSend::StorageItem* RtpStreamSend::StorageItemBuffer::Get(uint16_t seq) const
+	{
+		auto idx{ static_cast<uint16_t>(seq - this->startSeq) };
```

Comment:
Could this produce a negative value that would then be cashed to a false positive?

Reply:
Handled. We now check is `seq` is lower than` this->startSeq` and return `nullptr` in that case.